src/pages/workspaces/WorkspacesList/WorkspacesList.ts

```diff
@@ -0,0 +1,112 @@
+import _ from 'lodash/fp';
+import { ReactNode, useMemo, useState } from 'react';
+import { div, h, p } from 'react-hyperscript-helpers';
+import { Link, topSpinnerOverlay, transparentSpinnerOverlay } from 'src/components/common';
+import FooterWrapper from 'src/components/FooterWrapper';
+import { icon } from 'src/components/icons';
+import TopBar from 'src/components/TopBar';
+import { Ajax } from 'src/libs/ajax';
+import { isAzureUser } from 'src/libs/auth';
+import { withErrorIgnoring } from 'src/libs/error';
+import { updateSearch, useRoute } from 'src/libs/nav';
+import { useOnMount } from 'src/libs/react-utils';
+import { elements as StyleElements } from 'src/libs/style';
+import { newTabLinkProps } from 'src/libs/utils';
+import { cloudProviderTypes, WorkspaceWrapper as Workspace } from 'src/libs/workspace-utils';
+import { catagorizeWorkspaces } from 'src/pages/workspaces/WorkspacesList/CatagorizedWorkspaces';
+import { RecentlyViewedWorkspaces } from 'src/pages/workspaces/WorkspacesList/RecentlyViewedWorkspaces';
+import { useWorkspacesWithSubmissionStats } from 'src/pages/workspaces/WorkspacesList/useWorkspacesWithSubmissionStats';
+import {
+  getWorkspaceFiltersFromQuery,
+  WorkspaceFilters,
+  WorkspaceFilterValues,
+} from 'src/pages/workspaces/WorkspacesList/WorkspaceFilters';
+import { WorkspacesListModals } from 'src/pages/workspaces/WorkspacesList/WorkspacesListModals';
+import { WorkspacesListTabs } from 'src/pages/workspaces/WorkspacesList/WorkspacesListTabs';
+import {
+  WorkspaceUserActions,
+  WorkspaceUserActionsContext,
+} from 'src/pages/workspaces/WorkspacesList/WorkspaceUserActions';
+
+export const persistenceId = 'workspaces/list';
+
+export const getWorkspace = (id: string, workspaces: Workspace[]): Workspace =>
+  _.find({ workspace: { workspaceId: id } }, workspaces)!;
+
+export const WorkspacesList = (): ReactNode => {
+  const {
+    workspaces,
+    refresh: refreshWorkspaces,
+    loadingWorkspaces,
+    loadingSubmissionStats,
+  } = useWorkspacesWithSubmissionStats();
+
+  const [featuredList, setFeaturedList] = useState<Workspace[]>();
```

Comment:
`featuredList` is not a `Workspace[]`. All the featured workspaces file contains is namespaces and names.